src/Confluent.Kafka/Producer.cs

```diff
@@ -675,14 +676,29 @@ public int Flush()
         /// </remarks>
         public void Dispose()
         {
+            // TODO: If this method is called in a finalizer, can callbackTask potentially be null?
```

Comment:
should you add a test for that case?

Reply:
yes. this is an old todo. lower priority.